content/news/buzz/2022q3-roadmap/contents.lr

```diff
@@ -0,0 +1,107 @@
+
+* Adding support for linuxdeploy plugins for our existing AppImage backend; and
+
+* Adding a Flatpak backend.
+
+Windows packaging
+~~~~~~~~~~~~~~~~~
+
+Briefcase has an MSI packaging backend for Windows; and while this backend works, it doesn't deliver an end-user experience that is as polished as we'd like. We should be starting initial work on Windows packaging in the coming week, with the complete solution completed in early Q3. This will include:
+
+* Adding a stub binary for Windows apps, so that Briefcase apps present to the task manager with the apps name and icon, rather than "pythonw";
+
+* Improving log capture on Windows, so that it is easier to diagnose when problems occur; and
+
+* Fixing some issues with the way the `site` is defined and used. This may also address some issues we've seen with binary modules in some Windows apps.
+
+Binary modules
+~~~~~~~~~~~~~~
+
+The biggest caveat on BeeWare's mobile app story has always been that we don't provide support for binary packages on mobile platforms. In Q3, we're going to start addressing this limitation.
+
+There are already tools like `crossenv <https://github.com/benfogle/crossenv>`__ that can simplify this process, and we've got `a PR in our review queue <https://github.com/beeware/briefcase/pull/647>`__ that provides a proof-of-concept implementation of Android binary packaging. We need to take these tools, plus other tools in the Python ecosystem, and extend Briefcase so that we can present a clear development story for mobile developers that want to use modules with binary components.
+
+On iOS, there is an additional complication - we need to resolve how to manage dynamic loading of modules. BeeWare's iOS support currently relies on static complication of all binary modules. This was originally done because prior to iOS8, dynamic linking was prohibited on iOS. The days of iOS8 are now long behind us; but the Apple support package has retained this old approach, mostly because it wasn't clear what approach Apple would allow through the App Store review process. We will be making changes to Briefcase's iOS support to accomodate dynamic module loading, and then extending that support to include user-provided binary modules.
+
+Testing (especially GUI testing)
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Without testing, it's very difficult to ensure the stability and reliability of code in the long term. Unfortunately, many of the areas where BeeWare operates are inherently difficult to test. Visual GUI tests are notoriously difficult to perform; packaging tests require the ability actually install and run finished applications; testing mobile applications requires access to mobile device hardware in a CI setup. As a project, we've avoided writing tests for many of our core features due to these complexities - and we've paid the price for this in regressions. Now that we have more time, we can focus on setting up the frameworks to let us perform this testing.
```

Comment:
typo

> packaging tests require the ability actually install and run finished applications

packaging tests require the ability to actually install and run finished applications